### PR TITLE
#1547 Remove == operator for PassPhrase and KeyDetails

### LINKED ISSUE #1547: get rid of == override for KeyDetails
There are three ways to possibly compare key details:
 - is the fingerprint the same?
 - is the fingerprint and last modified the same?
 - is all of the information the same?

Whenever one uses `==` I'd rather that the comparison was explicit like `keyDetails1.id == keyDetails2.id` or similar.

Therefore I'd like to remove 

```swift
    static func == (lhs: KeyDetails, rhs: KeyDetails) -> Bool {
        lhs.ids == rhs.ids
    }
```

To prevent unnecessary surprises.

### PR DESCRIPTION
**Pull request by ioanmo226**
This PR removed == operator for PassPhrase and KeyDetails

close #1547 // if this PR closes an issue

----------------------------------

**Tests** _(delete all except exactly one)_:
- Does not need tests (refactor only, docs or internal changes)

--------------------------------

### To be filled by reviewers

I have reviewed that this PR... _(tick whichever items you personally focused on during this review)_:
- [ ] addresses the issue it closes (if any)
- [ ] code is readable and understandable
- [ ] is accompanied with tests, or tests are not needed
- [ ] is free of vulnerabilities
- [ ] is documented clearly and usefully, or doesn't need documentation

### CODE REVIEW

File: FlowCrypt/Core/Models/KeyAlgo.swift [hunk 1]
```diff
@@ -14,3 +14,9 @@ struct KeyAlgo: Decodable {
     let bits: Int?
     let curve: String?
 }
+
+extension KeyAlgo: Equatable {
+    static func == (lhs: KeyAlgo, rhs: KeyAlgo) -> Bool {
+        lhs.algorithmId == rhs.algorithmId
```
Comment (tomholub) [inline FlowCrypt/Core/Models/KeyAlgo.swift:20]: Should equate all fields. The curve may still be different, amount of bits may still be different, etc.

**Comment by tomholub** [comment]
`keyDetails1 == keyDetails2` should mean all fields are different, in either case.

You should check where this `Set(listOfKeyDetails)` is used, and make sure that they are unique in another way.

As implemented, it's too obviousness where such functionality is employed which is not good.

**Comment by ioanmo226** [comment]
To make `Set` from struct, it requires struct to confirm `Hashable` protocol and `Hashable` protocol requires to confirm to `Equatable` Protocol.
Therefore, we had custom `==` operator which we compare fingerprints(ids) of KeyDetails.

I don't think we can make `Set` without confirming to `Hashable` protocol.

**Comment by tomholub** [comment]
It's used here:

```swift
        let matchingKeyBackups = Set(try await keyMethods.filterByPassPhraseMatch(keys: backups, passPhrase: passPhrase))
```

`Set` by only fingerprint is silly here, anyway. If there are two keys with the same fingerprint, should choose the version that has newer `lastModified`. You can write a method for this: `keyDetailsArray.getUniqueByFingerprintByPreferingLatestLastModified()` and then you could unit-test that method instead.

**Comment by tomholub** [comment]
> I don't think we can make Set without confirming to Hashable protocol.

We don't need to use a `Set` at all. It was apparently only used to make the entries unique by fingerprint, which anyway was not the right thing to do, as implemented.

**Comment by ioanmo226** [comment]
Now, I got you. Then let me get rid of `Set` here and implement your proposed solution.

File: FlowCrypt/Controllers/Setup/SetupManuallyEnterPassPhraseViewController.swift [hunk 2]
```diff
@@ -238,8 +238,12 @@ extension SetupManuallyEnterPassPhraseViewController {
     }
 
     private func importKeys(with existedKeys: [KeyDetails], and passPhrase: String) throws {
```
Comment (tomholub) [inline FlowCrypt/Controllers/Setup/SetupManuallyEnterPassPhraseViewController.swift:240]: `existedKeys` should be called `existingKeys` :-) thanks

Comment (ioanmo226) [reply FlowCrypt/Controllers/Setup/SetupManuallyEnterPassPhraseViewController.swift]: ok

**Comment by tomholub** [comment]
Will let Roma review

File: FlowCrypt/Functionality/Services/Local Private Key Services/CombinedPassPhraseStorage.swift [hunk 3]
```diff
@@ -35,12 +35,13 @@ struct PassPhrase: Codable, Hashable, Equatable {
...
-    // itself, and not by primary fingerprint of the associated key. I understand this is being used somewhere,
-    // but I suggest to refactor it to avoid defining this == overload.
+    // We still need == operator here because we use `withUpdatedDate` to set `date` field to up-to-date
+    // Therfore, 2 passphrases might be treated differently even though
+    // they are exactly same if we don't implement custom == operator
     static func == (lhs: PassPhrase, rhs: PassPhrase) -> Bool {
-        lhs.primaryFingerprintOfAssociatedKey == rhs.primaryFingerprintOfAssociatedKey
+        return lhs.primaryFingerprintOfAssociatedKey == rhs.primaryFingerprintOfAssociatedKey
+                || lhs.value == rhs.value
+                || lhs.email == rhs.email
```
Comment (sosnovsky) [inline FlowCrypt/Functionality/Services/Local Private Key Services/CombinedPassPhraseStorage.swift:44]: Is it okay that passphrases will be equal even when they have different `primaryFingerprintOfAssociatedKey` but the same `email`?

File: FlowCrypt/Core/Models/KeyDetails.swift [hunk 4]
```diff
@@ -92,4 +84,18 @@ extension Array where Element == KeyDetails {
     var joinedPrivateKey: String {
         compactMap(\.private).joined(separator: "\n")
     }
+    
+    func getUniqueByFingerprintByPreferingLatestLastModified() -> [KeyDetails] {
+        var newArray: [KeyDetails] = []
```
Comment (sosnovsky) [inline FlowCrypt/Core/Models/KeyDetails.swift:89]: It'll be better to use more meaningful array name here, like `uniqueKeyDetails`

File: FlowCrypt/Functionality/Services/Local Private Key Services/CombinedPassPhraseStorage.swift (hunk 3 above)
Comment (ioanmo226) [reply FlowCrypt/Functionality/Services/Local Private Key Services/CombinedPassPhraseStorage.swift:44]: OHhh, no. wrong implementation.
Should be `&&`, not `||`.
Thanks for your notice

**Pull request merged**